AlphaWallet/Tokens/Coordinators/GetIsERC1155ContractCoordinator.swift

```diff
@@ -10,17 +10,26 @@ import web3swift
 
 class GetIsERC1155ContractCoordinator {
     private let server: RPCServer
+    private var cache: CachedERC1155ContractDictionary?
 
     private struct ERC165Hash {
         //https://eips.ethereum.org/EIPS/eip-1155
         static let official = "0xd9b67a26"
     }
-    init(forServer server: RPCServer) {
+    init(forServer server: RPCServer, cacheName: String = "ERC1155ContractCache.json") {
         self.server = server
+        cache = CachedERC1155ContractDictionary.loadFromCache(fileName: cacheName)
     }
 
     func getIsERC1155Contract(for contract: AlphaWallet.Address) -> Promise<Bool> {
-        return GetInterfaceSupported165Coordinator(forServer: server)
-            .getInterfaceSupported165(hash: ERC165Hash.official, contract: contract)
-    } 
+        if let result = cache?.isERC1155Contract(for: contract) {
+            return Promise.value(result)
+        }
+        return firstly { GetInterfaceSupported165Coordinator(forServer: server)
```

Comment:
Can you format this:

```
return firstly { GetInterfaceSupported165Coordinator(forServer: server)
        .getInterfaceSupported165(hash: ERC165Hash.official, contract: contract)
}
.get { result in
    self.cache?.setContract(for: contract, result)
}
```

like this instead?

```
return firstly {
    GetInterfaceSupported165Coordinator(forServer: server).getInterfaceSupported165(hash: ERC165Hash.official, contract: contract)
}.get { result in
    self.cache?.setContract(for: contract, result)
}
```

The indented parts show the async/promise-based code